### PR TITLE
GBM Trigger Ex

### PR DESCRIPTION
**Pull request by Vidushi-GitHub**
Kindly review this GBM Trigger example.

I had to change the type of statistics.schema/"classification" to string, number was wrong way to define it.

### CODE REVIEW

**Comment by Tohuvavohu** [comment]
Since classification is expecting a:

> "description": "Dictionary mapping mutually exclusive source classes to probabilities between 0 and 1, the sum of all values must be 1. e.g. ({'BNS', 0.9}, {'NSBH', 0.05}, {'BBH', 0.05})"


This is why Dakota and I set it to number in https://github.com/nasa-gcn/gcn-schema/pull/40
Was this a mistake? I don't follow.

**Comment by Vidushi-GitHub** [comment]
Yes, I saw you kept it as "GRB":1, but according to the description, it looks like GW merger-type probability. Should not it be enum + number, or string?

**Comment by Tohuvavohu** [comment]
That's an example, in the GUANO example case the classification is GRB.
For Fermi/GBM onboardd triggers it could be eg:
 ({'GRB', 0.8}, {'SFLARE', 0.1}, {'SGR',0.1})

The value *must* be a number if the normalization requirement is to be met. The key should be a string.

I'm not sure how to specify this, but the type:object, additional properties type:number seemed to work when we validated in https://github.com/nasa-gcn/gcn-schema/pull/40

@dakota002 or Leo probably have a better opinion

**Comment by jracusin** [comment]
I think in this format it can be read in as a python dictionary.  We borrowed this from LVK, but can be used for other sorts of classification.

**Comment by Tohuvavohu** [comment]
I agree, but note the LVK has the values and numbers not strings, which is my point:

>         "classification": {
            "BNS": 0.95,
            "NSBH": 0.01,
            "BBH": 0.03,
            "Terrestrial": 0.01
        },
^https://emfollow.docs.ligo.org/userguide/content.html

File: gcn/notices/core/EnergyRange.schema.json [hunk 1]
```diff
@@ -16,16 +16,18 @@
...
       "enum": ["keV", "nm", "Hz"],
       "description": "Units of band range, if not parsed, then default energy is keV",
-      "type": "string"
+      "type": "string",
+      "default": "keV"
     },
     "filter": {
       "type": "string",
-      "description": "Filter name, as used in optical observations"
+      "description": "Optional filter name, as used in optical observations"
```
Comment (jracusin) [inline gcn/notices/core/EnergyRange.schema.json:30]: Should this be an array?  Instruments often report observations in multiple filters.

File: gcn/notices/fermi/gbm/Trigger.example.json [hunk 2]
```diff
@@ -1,82 +1,44 @@
...
+
+  "event_name": ["GRB 221009A", "Swift J1913.1+1946"],
+  "id": ["bn 221009553"],
+
+  "trigger_time": "2022-10-09T13:16:59.000Z",
+
+  "ra": 290.4,
+  "dec": 22.3,
+  "semi_major": [1],
+  "healpix_url": "glg_healpix_all_bn221009553_v01",
```
Comment (jracusin) [inline gcn/notices/fermi/gbm/Trigger.example.json:21]: That's not really a URL, but that is also a limitation in GCN Classic notices.  Here's the URL for this file: https://heasarc.gsfc.nasa.gov/FTP/fermi/data/gbm/bursts/2022/bn221009553/current/glg_healpix_all_bn221009553_v01.fit

File: gcn/notices/core/EnergyRange.schema.json [hunk 3]
```diff
@@ -16,16 +16,18 @@
...
     },
     "units": {
       "enum": ["keV", "nm", "Hz"],
       "description": "Units of band range, if not parsed, then default energy is keV",
-      "type": "string"
+      "type": "string",
+      "default": "keV"
     },
     "filter": {
       "type": "string",
```
Comment (Vidushi-GitHub) [inline gcn/notices/core/EnergyRange.schema.json:29]:
```suggestion
      "type": "array",
```

Comment (Vidushi-GitHub) [inline gcn/notices/core/EnergyRange.schema.json:29]:
```suggestion
      "type": "array",
```

File: gcn/notices/fermi/gbm/Trigger.example.json (hunk 2 above)
Comment (Vidushi-GitHub) [reply gcn/notices/fermi/gbm/Trigger.example.json:21]:
```suggestion
  "healpix_url": "https://heasarc.gsfc.nasa.gov/FTP/fermi/data/gbm/bursts/2022/bn221009553/current/glg_healpix_all_bn221009553_v01.fit",
```

File: gcn/notices/icecube/GoldAndBronze.example.json [hunk 4]
```diff
@@ -45,7 +45,7 @@
     },
     "p_astro": 0,
     "classification": {
-      "type": 0
+      "type": "0"
```
Comment (Vidushi-GitHub) [inline gcn/notices/icecube/GoldAndBronze.example.json:48]:
```suggestion
      "GRB": 1
```

File: gcn/notices/swift/bat/guano/loc-map.example.json [hunk 5]
```diff
@@ -26,7 +26,7 @@
       "band": "energy",
       "units": "keV"
     },
-    "classification": { "GRB": 1 }
+    "classification": { "GRB": "1" }
```
Comment (Vidushi-GitHub) [inline gcn/notices/swift/bat/guano/loc-map.example.json:29]:
```suggestion
    "classification": { "GRB": 1 }
```

File: gcn/notices/swift/bat/guano/loc-arcmin.example.json [hunk 6]
```diff
@@ -37,7 +37,7 @@
       "band": "energy",
       "units": "keV"
     },
-    "classification": { "GRB": 1 }
+    "classification": { "GRB": "1" }
```
Comment (Vidushi-GitHub) [inline gcn/notices/swift/bat/guano/loc-arcmin.example.json:40]:
```suggestion
    "classification": { "GRB": 1 }
```

File: gcn/notices/swift/bat/guano/alert.example.json [hunk 7]
```diff
@@ -22,7 +22,7 @@
       "band": "energy",
       "units": "keV"
     },
-    "classification": { "GRB": 1 }
+    "classification": { "GRB": "1" }
```
Comment (Vidushi-GitHub) [inline gcn/notices/swift/bat/guano/alert.example.json:25]:
```suggestion
    "classification": { "GRB": 1}
```

File: gcn/notices/fermi/gbm/Trigger.example.json [hunk 8]
```diff
@@ -1,82 +1,44 @@
...
+
+  "far": 0.1,
+  "rate_snr": 250,
+  "rate_duration": 37,
+  "image_snr": 150,
+  "image_duration": 22.3,
+  "energy_low": 9,
+  "energy_high": 30000,
+  "p_astro": 0.99,
+  "classification": { "type": "{'BNS', 0.9}" },
```
Comment (Vidushi-GitHub) [inline gcn/notices/fermi/gbm/Trigger.example.json:31]:
```suggestion
  "classification": { "GRB": 0.99, "Flare": 0.01},
```

File: gcn/notices/core/Statistics.schema.json [hunk 9]
```diff
@@ -35,11 +35,11 @@
     },
     "p_astro": {
       "type": "number",
-      "description": "Probability that source is of astrophysical origin"
+      "description": "Probability [dimensionless, 0-1] that source is of astrophysical origin"
     },
     "classification": {
       "type": "object",
-      "additionalProperties": { "type": "number" },
+      "additionalProperties": { "type": "string" },
```
Comment (Vidushi-GitHub) [inline gcn/notices/core/Statistics.schema.json:42]:
```suggestion
      "additionalProperties": { "type": "number" },
```

**Comment by Vidushi-GitHub** [comment]
@Tohuvavohu we have kept the classification as it is. Thanks!

**Pull request merged**